File: Solutions/Easy/744. Find Smallest Letter Greater Than Target/Solution.cs
public class Solution {
    public char NextGreatestLetter(char[] letters, char target) {
        int letterPos = -1;
        for(int i = 0; i < letters.Length; i++)
        {
            letterPos = letters[i] > target ? (letterPos == -1 ? i : (letters[letterPos] > letters[i] ? i : letterPos))  : letterPos;
        }
        return letterPos == -1 ? letters[0] : letters[letterPos];
    }
}

File: Solutions/Easy/744. Find Smallest Letter Greater Than Target/Solution.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        result = "zz"
        for c in letters:
            if (c <= target):
                continue
            if (c < result):
                result = c
        if (result == "zz"):
            return letters[0]
        return result